### PR TITLE
Refactored create channel tests to wait for notification

### LINKED ISSUE #2277: Error when running tests on latest devel
When running the tests on Mac (locally), the following error occurred (at random):

```
Traceback (most recent call last):
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/Library/Python/2.7/site-packages/twisted/_threads/_threadworker.py", line 46, in work
    task()
  File "/Library/Python/2.7/site-packages/twisted/_threads/_team.py", line 190, in doWork
    task()
--- <exception caught here> ---
  File "/Library/Python/2.7/site-packages/twisted/python/threadpool.py", line 246, in inContext
    result = inContext.theWork()
  File "/Library/Python/2.7/site-packages/twisted/python/threadpool.py", line 262, in <lambda>
    inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
  File "/Library/Python/2.7/site-packages/twisted/python/context.py", line 118, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/Library/Python/2.7/site-packages/twisted/python/context.py", line 81, in callWithContext
    return func(*args,**kw)
  File "/Users/martijndevos/Documents/tribler_endpoints/Tribler/Core/Modules/channel/channel.py", line 85, in _on_channel_created
    if channel_data[u'channel'].cid != self._channel_community.cid:
exceptions.AttributeError: 'NoneType' object has no attribute 'cid'
```

We've also seen this one in at least one other PR (#2274) during the test execution. I will fix it as separate commit in my PR #2253.

### PR DESCRIPTION
**Pull request by devos50**
Fixes #2277.

### CODE REVIEW

**Comment by devos50** [comment]
@lfdversluis can you review?

File: Tribler/Test/Core/test_session.py [hunk 1]
```diff
@@ -85,38 +87,49 @@ def setUp(self, autoload_discovery=True):
...
-            self.session.add_torrent_def_to_channel(channel_id, torrent_def, extra_info, forward=False)
-        do_test()
+        def on_channel_created(subject, change_type, object_id, channel_data):
+            channel_id = self.channel_db_handler.getMyChannelId()
+            self.session.add_torrent_def_to_channel(channel_id, torrent_def, {"description": "iso"}, forward=False)
+            self.assertTrue(self.channel_db_handler.hasTorrent(channel_id, torrent_def.get_infohash()))
+            test_deferred.callback(None)
+
+        self.session.add_observer(on_channel_created, SIGNAL_CHANNEL, [SIGNAL_ON_CREATED])
+        self.session.create_channel("name", "description", "open")
```
Comment (lfdversluis) [inline Tribler/Test/Core/test_session.py]: That's a really funny way to wait for a notifier to fire

File: Tribler/Test/Core/test_session.py [hunk 2]
```diff
@@ -85,38 +87,49 @@ def setUp(self, autoload_discovery=True):
...
         @blocking_call_on_reactor_thread
-        def do_test():
-            self.session.add_torrent_def_to_channel(channel_id, torrent_def, forward=False)
-            self.session.add_torrent_def_to_channel(channel_id, torrent_def, forward=False)
-        do_test()
+        def on_channel_created(subject, change_type, object_id, channel_data):
+            channel_id = self.channel_db_handler.getMyChannelId()
+            try:
+                self.session.add_torrent_def_to_channel(channel_id, torrent_def, forward=False)
+                self.session.add_torrent_def_to_channel(channel_id, torrent_def, forward=False)
```
Comment (lfdversluis) [inline Tribler/Test/Core/test_session.py]: maybe invoke the errback here?

**Comment by lfdversluis** [comment]
Comments given!

File: Tribler/Test/Core/test_session.py (hunk 2 above)
Comment (devos50) [inline Tribler/Test/Core/test_session.py]: If I invoke the errback here, the deferred fails right (and so the test)?

Comment (lfdversluis) [inline Tribler/Test/Core/test_session.py]: It should since we had deferreds erroring in the past but tests reporting ok. I believe we fixed that, @whirm ?

Comment (whirm) [inline Tribler/Test/Core/test_session.py]: If you decorate a test with @deferred  and return a deferred that fails, nose will catch it. If a random deferred fails and nothing catches the failure, our checks will catch it and fail the test.

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
```
test_startup_session_save_load (Tribler.Test.Core.test_sessionconfig.TestSessionConfig) ... ok

======================================================================
ERROR: test_get_tdef (Tribler.Test.Core.Libtorrent.test_libtorrent_download_impl.TestLibtorrentDownloadImpl)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\workspace\jenkins\workspace\GH_Tribler_PR_tests_win64\tribler\Tribler\Test\test_as_server.py", line 281, in tearDown
    super(TestAsServer, self).tearDown(annotate=False)
  File "C:\workspace\jenkins\workspace\GH_Tribler_PR_tests_win64\tribler\Tribler\Core\Utilities\twisted_thread.py", line 174, in wrapper
    % timeout)
TimeExpired: timeout expired before end of test (5.000000 s.)
```

That bug is so weird...

**Comment by devos50** [comment]
@whirm ready for merge!

**Pull request merged**